Add windows support

tadeu opened the pull request.

tadeu committed:
Add windows support

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

tadeu commented on recipe/meta.yaml:
```diff
@@ -59,3 +71,5 @@ about:
   recipe-maintainers:
     - isuruf
+    - tadeu
+    - doxygen
```
@doxygen Are you ok being added as a co-maintainer of this conda-forge package? This will give you read/write/merge privileges on the resulting feedstock repository.

tadeu replied:
@doxygen Since you didn't answer, I'm removing your name from the list in order to get this PR ready for merging, but feel free to join at any time :)

isuruf commented on recipe/meta.yaml:
```diff
@@ -1,40 +1,52 @@
+      - copy doxyindexer.exe %SCRIPTS%\ || goto :eof       # [win]
+      - copy doxysearch.cgi.exe %SCRIPTS%\ || goto :eof    # [win]
+      - copy libclang.dll %LIBRARY_BIN%\ || goto :eof      # [win]
+      - copy libclang.dll %LIBRARY_BIN%\ || goto :eof      # [win]
```
I'd rather not depend on a dll like this as this might interfere with another package providing `libclang.dll`.
Is it hard to build on windows? If so, then I'm okay with repackaging the binary.

tadeu replied:
I'll see if it's possible to build on Windows with `libclang` statically linked.

tadeu replied:
@isuruf It's easy to build it on Windows, but _without clang support_. There are no precompiled _static_ version of `libclang` on Windows.

Unfortunately, Doxygen depends on clang for "a more accurate source browsing experience, and better syntax highlighting, cross referencing, and call graphs" [(ref.)](http://lists.llvm.org/pipermail/cfe-dev/2013-May/029732.html).

I agree this could be a problem, but I think there are no other packages using `libclang.dll`, and there are no `libclang` packages for Windows. Whenever `libclang` or `clang` is added to conda-forge, we can update this recipe to depend on it. What do you think?

isuruf replied:
Sure, we can have it as it is until, https://github.com/conda-forge/staged-recipes/pull/1481 is merged.

isuruf reviewed (commented):
You need to rerender for a change like this.

ocefpaf commented on recipe/meta.yaml:
```diff
@@ -1,40 +1,52 @@
+  
 build:
   number: 0
-  skip: True  # [win or not py35]
```
If you are removing the Windows `skip` you need to keep the `vc`s.
See http://conda-forge.github.io/docs/buildwin.html#vc-features

isuruf replied:
I don't think you need them. This is just repackaging binaries.

ocefpaf replied:
I missed that. In that case you are correct.

tadeu committed:
MNT: Re-rendered with conda-smithy 1.7.0

jakirkham committed:
Hide the `requirements` on Windows

Sometimes having an empty `requirements` section, like with this one on
Windows, results in strange behavior. Particularly when re-rendering.
Here we cover up the rest of this section with selectors so that it
vanishes on Windows.

jakirkham committed:
Only require a Python version on Unix

Requiring a Python version on Windows when no Python is installed is
paramount to skipping Windows outright as the requirement cannot be
satisfied. This fixes that problem by only requiring a Python version be
matched on Unix where Python is used.

jakirkham commented:
Added PR ( https://github.com/tadeu/doxygen-feedstock/pull/1 ) against your fork to re-render this PR in response to the issues you had raised, @tadeu . Please take a look.

tadeu commented:
Thanks @jakirkham. I had already fixed the problem manually (I rerendered removing the `skip` line, then put the skip line back again), so I think these changes won't be needed.

tadeu committed:
Change recipe maintainers

tadeu commented:
@isuruf Can it be merged? Thanks!

isuruf commented:
I'll merge this once the tests pass.

jakirkham commented on recipe/LICENSE:
```diff
@@ -0,0 +1,339 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Why do we need this license file?

jakirkham commented:
> Thanks @jakirkham. I had already fixed the problem manually (I rerendered removing the `skip` line, then put the skip line back again), so I think these changes won't be needed.

So the problem with this is that the re-rendering bot or another user might make the same mistake. 

Also there is another change in my PR to make sure the team update script doesn't choke on this recipe when parsing through it for maintainers.

Could you please a similar change to your PR or merge mine? Either way is fine with me.

tadeu committed:
Merge pull request #1 from jakirkham/fix_enable_win

Enable Windows

tadeu commented:
@isuruf Could this be merged now? Thanks!

isuruf commented:
Thanks for the PR, @tadeu.
